subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -672,21 +736,65 @@ int ztls_poll(struct zsock_pollfd *fds, int nfds, int timeout)
 int ztls_getsockopt(int sock, int level, int optname,
 		    void *optval, socklen_t *optlen)
 {
+	int err;
+	struct net_context *context = INT_TO_POINTER(sock);
+
 	if (level != SOL_TLS) {
 		return zsock_getsockopt(sock, level, optname, optval, optlen);
 	}
 
-	errno = -ENOPROTOOPT;
-	return -1;
+	if (!context || !context->tls) {
+		return -EBADF;
+	}
+
+	if (!optval || !optlen) {
+		return -EFAULT;
+	}
+
+	switch (optname) {
+	case TLS_SEC_TAG_LIST:
+		err =  tls_opt_sec_tag_list_get(context, optval, optlen);
+		break;
+
+	default:
+		err = -ENOPROTOOPT;
```

Comment:
Even thou it makes no difference here, I would recommend that there is a break; statement also in the default branch. It is a good habit to do and avoids errors if new cases are added later.